### DISCUSSION TITLE
Asyncio import csv

### DISCUSSION THREAD

**Discussion by wnesbv** [sqlalchemy/sqlalchemy#8387] [Usage Questions]
Good afternoon. Share your experience who is familiar with the situation: only the first line from the csv file is added.. And existing rows are not updated because of uniqueness..
`  

    async with aiofiles.open(
        directory, mode="r", encoding="utf-8", newline=""
    ) as afp:
        async for new in AsyncDictReader(afp):
            query = ScheduleService(
                **{
                    "id": new["id"],
                    "name": new["name"],
                    "type": new["type"],
                    "title": new["title"],
                    "description": new["description"],
                    "date": datetime.strptime(
                        new["date"], settings.DATE
                    ).date(),
                    "there_is": datetime.strptime(
                        new["there_is"], settings.DATETIME_FORMAT
                    ),
                    "created_at": datetime.now(),
                    "sch_s_owner": new["sch_s_owner"],
                    "sch_s_service_id": new["sch_s_service_id"],
                }
            )
            # ..
            session.add(query)
            session.refresh([query])
            await session.commit()
`

**Comment by CaselIT** [comment]
Hi,

I'm not sure I understand if there is question or if it's just a statement

**Reply by wnesbv** [reply]
https://github.com/wnesbv/starlette-admin-auth-blog-booking/blob/main/item/schedule_export_csv.py
https://github.com/wnesbv/starlette-admin-auth-blog-booking/blob/main/item/schedule_import_csv.py

**Reply by CaselIT** [reply]
Sorry but I'm still not sure if there is you are trying to showcase a problem or not...

**Reply by wnesbv** [reply]
Problems in orm. Orm is required to have a memory dump (import export), without a memory dump is: so a trifle, a laboratory rat - played and forgot..

**Comment by jvanasco** [comment]
No one will be able to help you without a SSCCE "Short, Self Contained, Correct (Compilable), Example" see http://sscce.org/ 
These are also known as a MCVE "Minimal, Reproducible Example" (https://stackoverflow.com/help/minimal-reproducible-example)

Please try to create and share a single file example that people can run without downloading an entire project.


Here is a listing of examples from this project:  https://github.com/sqlalchemy/sqlalchemy/issues?q=label%3A%22great+mcve%22+is%3Aclosed
